src/java.base/share/classes/sun/invoke/util/Wrapper.java

```diff
@@ -376,6 +380,9 @@ public static Wrapper forBasicType(Class<?> type) {
         }
     }
 
+    /** A nominal descriptor of the primitive type */
+    public ClassDesc primitiveClassDescriptor() { return primitiveTypeDesc; }
```

Comment:
While I agree it may seem non-sensical to define `CD_Object` for `Object`, there's precedent in that `Wrapper.primitiveType` is `Object.class` - so for consistency I wired `primitiveTypeDesc` of `OBJECT` to `CD_Object`. For the usage patterns introduced with this patch we'll never call `primitiveClassDescriptor()` on `Wrapper.OBJECT`, though, so we can revisit this. 
 
`void.class` is the primitive type for `Void.class`, allowed as a return type and recognized by `Wrapper.forPrimitiveType`. There are checks in for example `ConstantUtils.parseMethodDescriptor` which disallows `void.class` anywhere but as a return type.

Reply:
I know it'll never call `primitiveClassDescriptor()` besides primitive types.   While `void.class.isPrimitive()` returns true, `void` is not a primitive type and called out explicitly in the spec.

My point is about the behavior does not match the method name.  It may be better to rename it to `classDescriptor` instead.  It's okay to clean up in your other performance fix (perhaps #18945).

Reply:
Updated #18945, will try ignoring the pre-existing `Wrapper.primitiveType` inconsistencies.